### ISSUE TITLE
Document TokenContract

### ISSUE THREAD

**Issue by nicc** [o1-labs/docs2#860]
In order to implement custom tokens on Mina
As an o1js developer
I want to know how to use the `TokenContract` class

**Acceptance Criteria**
_1) Remove the tutorial (we'll re-introduce a tutorial once a reference implementation is available)_
When I view the [zkApp Developers section](https://docs.minaprotocol.com/zkapps) of the docs
Then I do not see a custom tokens tutorial

_2) Update the existing Custom Tokens page_
When I view the [Custom Tokens page](https://docs.minaprotocol.com/zkapps/o1js/custom-tokens)
Then I see that all code snippets utilise the `TokenContract` class
And all code snippets work as describe
And I still see the list of important terms

_3) Overview_
When I view the [Custom Tokens page](https://docs.minaprotocol.com/zkapps/o1js/custom-tokens)
Then I see an introduction to custom tokens and their benefits
And this introduction is based on the preamble in the existing cutom tokens tutorial, as well as the first 1min and 30s of  [this video (draft)](https://www.youtube.com/watch?v=06EfQt1ZAU0) by Jack

_4) TokenContract class basics_
When I view the [Custom Tokens page](https://docs.minaprotocol.com/zkapps/o1js/custom-tokens)
Then I am guided through inheriting from the `TokenContract` class ([example](https://github.com/o1-labs/o1js/blob/main/src/lib/mina/token/token-contract.unit-test.ts#L13))
And I am shown how to mint tokens ([example](https://github.com/o1-labs/o1js/blob/main/src/lib/mina/token/token-contract.unit-test.ts#L29))
And I am shown how to burn tokens - _Reference / example pending... has this changed?_
And I am shown how to transfer tokens ([example](https://github.com/o1-labs/o1js/pull/1384/files#diff-89251d0951b2df2c4539287fa3c314198bfcc0c7835e3e8cd30d9e7c569979f3R60))
And I am shown how to implement the abstract `approveBase` method ([example](https://github.com/o1-labs/o1js/blob/main/src/lib/mina/token/token-contract.unit-test.ts#L17)), which replaces the currently documented approval mechanism
And I am shown how to transfer tokens where the token sending is approved by a smart contract (withdraw tokens from a contract) - _Example / reference pending_

_5) out of scope (these features will be documented in a new tutorial that depends on a reference implementation)_
When I view the [Custom Tokens page](https://docs.minaprotocol.com/zkapps/o1js/custom-tokens)
Then I am not guided through the permutations of [approvability](https://forums.minaprotocol.com/t/draft-fungible-token-standard-zkapps/6142#approvable-8)
And I do not see the outdated proof authorisation section currently on the custom tokens page
And I am not guided through the permutations of [administration](https://forums.minaprotocol.com/t/draft-fungible-token-standard-zkapps/6142#adminable-mintable-burnable-pausable-upgradable-11)
And I am not guided through the permutations of [viewability](https://forums.minaprotocol.com/t/draft-fungible-token-standard-zkapps/6142#viewable-13)
And I am not guided through the permutations of [hookability](https://forums.minaprotocol.com/t/draft-fungible-token-standard-zkapps/6142#hookable-16)

_Notes:_
- This serves to document in a basic form the `TokenContract` class that serves as a blueprint for custom token implementations
- It is based on a MIP proposal for a Fungible Token Standard and improves the underlying implementation to enable a better token standard
- As such, it is only a building block and not yet a full token standard
- Concurrently to this docs release, we'll update the proposed token standard to better utilise the TokenContract class, and build a reference implementation
- Once we have a reference implementation and udpated token standard, we'll reintroduce a more thoroough tutorial
- TL;DR: these are not the full docs, just a stepping stone

**Comment by mitschabaude** [comment]
> And I am shown how to transfer tokens where the token sending is approved by a smart contract (withdraw tokens from a contract) - Example / reference pending

It actually seems useful to include a simple example of this in the same token contract unit test. In the meantime, here is an example to reference: We call a DEX contract's `swap()` method, which (among other things) authorizes the spending of some Y tokens, and then we call `tokenY.transfer()` with _`dexY.self` as the sender_. The receiver is just an EOA specified by an address as usual.

https://github.com/o1-labs/o1js/blob/24537f740f82c2922a61e937b28fae8bf0fe068c/src/examples/zkapps/dex/dex-with-actions.ts#L199-L200

**Comment by barriebyron** [comment]
the `TokenContract` class enables the token standard

meanwhile, let's document how to use the lower level class before we have the reference implementation

**Comment by Pfed-prog** [comment]
I know it is very difficult to adapt to new token stndards moving from evm, I have created a governance forum post on Token Metadata https://forums.minaprotocol.com/t/mina-smart-contract-uri/6232
